### PR TITLE
Use longest match for resolver

### PR DESCRIPTION
**Pull request by mdboom**

### CODE REVIEW

**Comment by embray** [comment]
I don't fully understand the background to this issue.  Could somebody fill me in a bit more?

File: pyasdf/tests/test_asdftypes.py [hunk 1]
```diff
@@ -180,3 +181,27 @@ def url_mapping(self):
...
+        @property
+        def url_mapping(self):
+            return [('http://stsci.edu/schemas/asdf/core/',
+                     'FOOBAR/{url_suffix}')]
+
+    l = extension.AsdfExtensionList(
+        [extension.BuiltinExtension(), FancyComplexExtension()])
+
+    assert l.url_mapping(
+        'http://stsci.edu/schemas/asdf/core/asdf-1.0.0') == 'FOOBAR/asdf-1.0.0'
```
Comment (embray) [inline pyasdf/tests/test_asdftypes.py]: And why would we want an extension to be able to override the path to a core schema?

**Comment by embray** [comment]
@nden -- any idea about this?

**Comment by nden** [comment]
This was to fix a problem with passing more than one custom extensions to`AsdfFile`. It failed to open the file in some cases depending on the order of  the extensions.

**Comment by embray** [comment]
I don't quite understand what the test is supposed to demonstrate though.  It seems to also be demonstrating that a custom extension can override the paths to built-in schemas, which we *probably* don't want.  I also don't understand how the test is demonstrating a "longest match".

File: pyasdf/tests/test_asdftypes.py (hunk 1 above)
Comment (nden) [inline pyasdf/tests/test_asdftypes.py]: I think it may be simply testing the machinery of the mapping.

Comment (embray) [inline pyasdf/tests/test_asdftypes.py]: Maybe, but I think it exposes a possible loophole we'd probably prefer to close :)

**Pull request merged**